anastruct/fem/system_components/util.py

```diff
@@ -21,35 +21,65 @@ def ensure_single_hinge(
     node_id2
         Id of the node in the system
     """
-    if spring is not None and 0 in spring:
+    if spring is not None and 0 in spring.values():
         """
         Must be one rotational fixed element per node. Thus keep track of the hinges (k == 0).
         """
 
         for node_nr in range(1, 3):
-            if spring[node_nr] == 0:  # node is a hinged node
+            if node_nr in spring and spring[node_nr] == 0:  # node is a hinged node
                 if node_nr == 1:
                     node_id = node_id1
                 else:
                     node_id = node_id2
 
-                system.node_map[node_id].hinge = True
+                if (
+                    node_nr in spring.keys() and spring[node_nr] == 0
+                ):  # node is a hinged node
+                    if len(system.node_map[node_id].elements) > 0:
+                        hinges = []
+                        for el in system.node_map[node_id].elements.values():
+                            if node_id == el.node_id1:
+                                hinges.append(1 in el.springs and el.springs[1] == 0)
+                            elif node_id == el.node_id2:
+                                hinges.append(2 in el.springs and el.springs[2] == 0)
+                        pass_hinge = not all(hinges)
+                    else:
+                        pass_hinge = True
+                    if not pass_hinge:
+                        system.node_map[node_id].hinge = True
+                        del spring[node_nr]  # too many hinges at that element.
+
+                elif node_nr not in spring.keys() or (
```

Comment:
Is this not always True if previous statement evaluates to False? Could we replace this with else?

Reply:
Yes it is... This is why reviews are always helpful :-)